Samples/iOS-Swift/iOS-Swift-UITests/ProfilingUITests.swift

```diff
@@ -152,72 +137,104 @@ extension ProfilingUITests {
     func stopContinuousProfiler() {
         app.buttons["io.sentry.ios-swift.ui-test.button.stop-continuous-profiler"].afterWaitingForExistence("Couldn't find button to stop continuous profiler").tap()
     }
-     
+
+    enum ProfilingType {
+        case trace
+        case continuous // aka "continuous beta"
+        case ui // aka "continuous v2"
+    }
+
+    func performTest(profileType: ProfilingType, lifecycle: SentryProfileOptions.SentryProfileLifecycle? = nil) throws {
+        try launchAndConfigureSubsequentLaunches(shouldProfileThisLaunch: false, shouldProfileNextLaunch: true, profileType: profileType, lifecycle: lifecycle)
+        try launchAndConfigureSubsequentLaunches(terminatePriorSession: true, shouldProfileThisLaunch: true, shouldProfileNextLaunch: false, profileType: profileType, lifecycle: lifecycle)
+    }
+
+    fileprivate func setAppLaunchParameters(_ profileType: ProfilingUITests.ProfilingType, _ lifecycle: SentryProfileOptions.SentryProfileLifecycle?, _ shouldProfileNextLaunch: Bool) {
+        app.launchArguments.append(contentsOf: [
```

Comment:
`m`: I'm not a real fan of using methods with multiple parameters for tests. It harms the readability of the tests. When I look at one single test I need to jump around a lot in the file to find out how the test works. I think some code duplication is acceptable for tests, because when you look at one test case you know what's going on. Tests only need to change when the requirements change. They usually don't change often. Also, with this approach, you end up having logic in tests that can contain bugs, and then the tests don't test what you want. Tests can be repetitive but they should be stupid.

That's a personal preference and maybe a bigger discussion. This is up to you and shouldn't block the PR.

Reply:
I agree with the readability aspect. Usually I'm doing this because I want to make it easy to see that the entirety of the search space is covered, which is harder to see in individual functions. But I will look into unrolling them in separate work.